File: Graph/DFS/do.py
class Solution:
    def doDFS(self, adj, ans, visited, node):
        visited[node] = True
        ans.append(node)

        for neighbour in adj[node]:
            if not visited[neighbour]:
                self.doDFS(adj, ans, visited, neighbour)

    def dfs(self, adj):
        visited = [False] * len(adj)
        ans = []

        self.doDFS(adj, ans, visited, 0)

        return ans

File: Graph/DFS/do.cpp
#include <vector>
using namespace std;

class Solution {
  public:
    void doDFS(vector<vector<int>>& adj,vector<int>& ans,vector<bool>& visited,int v){
        if(visited[v])return;
        ans.push_back(v);
        visited[v] = true;
        for(int neighbour: adj[v]){
            if(!visited[neighbour]) doDFS(adj,ans,visited,neighbour);
        } 
    }
    
    vector<int> dfs(vector<vector<int>>& adj) {
        // Code here
        int totalVertices = adj.size();
        vector<bool>visited(totalVertices,false);
        vector<int>ans;
        doDFS(adj,ans,visited,0);
        return ans;
    }
};